File: 680-valid-palindrome-ii/valid-palindrome-ii.js
var validPalindrome = function(s) {
    let i = 0
    let j = s.length - 1

    function isPalindrome(word) {
        i = 0
        j = word.length - 1

        while (i < j) {
            if (word[i] != word[j]) return false
            i += 1
            j -= 1
        }
        return true
    }   

    while (i <= j) {
        if (s[i] != s[j]) {
            word1 = s.slice(0, i) + s.slice(i + 1)
            word2 = s.slice(0, j) + s.slice(j + 1)
            if (!isPalindrome(word1) && !isPalindrome(word2)) return false
        }
        i += 1
        j -= 1
    }
    return true

};

File: 680-valid-palindrome-ii/valid-palindrome-ii.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        
        def isPalindrome(string):
            i = 0
            j = len(string) - 1

            while (i < j):
                if (string[i] != string[j]):
                    return False
                i += 1
                j -= 1

            return True

        a = 0
        b = len(s) - 1

        while (a < b):
            if (s[a] != s[b]):
                return isPalindrome(s[:a] + s[a + 1:]) or isPalindrome(s[:b] + s[b + 1:])
            a += 1
            b -= 1

        return True
        